internal/seclang/directives.go

```diff
@@ -1046,6 +1046,87 @@ func updateTargetBySingleID(id int, variables string, options *DirectiveOptions)
 	return rp.ParseVariables(strings.Trim(variables, "\""))
 }
 
+// Description: Updates the action list of the specified rule(s).
+// Syntax: SecRuleUpdateActionById ID ACTIONLIST
+// ---
+// This directive will overwrite the action list of the specified rule with the actions provided in the second parameter.
+// It has two limitations: it cannot be used to change the ID or phase of a rule.
+// Only the actions that can appear only once are overwritten.
+// The actions that are allowed to appear multiple times in a list, will be appended to the end of the list.
+// The following example demonstrates how `SecAuditEngine` is used:
+// ```apache
+// SecRuleUpdateActionById 12345 "deny,status:403"
+// ```
+func directiveSecRuleUpdateActionByID(options *DirectiveOptions) error {
+	if len(options.Opts) == 0 {
+		return errEmptyOptions
+	}
+
+	idsOrRanges := strings.Fields(options.Opts)
+	length := len(idsOrRanges)
+	if length < 2 {
+		return errors.New("syntax error: SecRuleUpdateActionById id \"ACTION1,ACTION2,...\"")
+	}
+	// The last element is expected to be the actions(s)
+	actions := idsOrRanges[length-1]
+	for _, idOrRange := range idsOrRanges[:length-1] {
```

Comment:
nit: maybe you use the index in the for to skip last element rather than creating a new slice.